### PR TITLE
To Do List Crud

### PR DESCRIPTION
**Pull request by akbarsazish**
0- Configure linters.
1- Set up a new project with webpack.
2-  Create an index.html.
3- Creat module folder and added two files form.js and todos.js
4- Create an index.js file and set an array of some simple to do tasks (array of objects)
5- Each task object  contain three keys: a- description [string]. b- completed [bool]. c- index: [number].
6- Create a style.css and set rules for the To Do List.
7- In this project you can add items, edit items and remove it.

### CODE REVIEW

**Review by 1GORDON — COMMENTED** [review]
##
## Hi @akbarsazish ✋

Good job so far!

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

### This project will be marked as Invalid and won't affect the number of your weekly reviews.

- [ ] Kindly remove the live demo section from your readme file since this project is not deployed. This is a general requirement for all projects.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/form.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  if (todoInput.value === '') {
+    errorMessag.innerHTML = "it can't be blank!!!";
+  } else {
+    errorMessag.innerHTML = '';
+    saveTodo(todosListParent);
+  }
+};
+
+// submiting the form
+form.addEventListener('keydown', (event) => {
```
Comment (Ismailco) [inline src/modules/form.js:33]: - [ ] When adding a new task to the list the index should start from `1` in the local storage not `0`, kindly make sure to fix this issue so the app works as expected

Screenshot_20230623_000139

File: src/modules/form.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import { todoLists, displayTodoList } from './todos.js';
+
+const form = document.getElementById('todoForm');
+const todoInput = document.getElementById('newtodo');
+const errorMessag = document.getElementById('errorMessag');
+const todosListParent = document.getElementById('todo-list');
+
+// Define the removeTodo function
+// eslint-disable-next-line no-use-before-define
+const removeTodo = (index) => { // eslint-disable-line
```
Comment (Ismailco) [inline src/modules/form.js]: - [ ] When removing a task from the list the index should replace the old removed task, kindly consider fixing this so all the indexes are displayed correctly

Screenshot_20230623_000204

File: src/modules/form.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  todoLists.splice(index, 1);
+  displayTodoList();
+  localStorage.setItem('todoLists', JSON.stringify(todoLists));
+};
+
+// add event listener to editBtn
+todosListParent.addEventListener('click', (e) => {
```
Comment (Ismailco) [inline src/modules/form.js]: - [ ] The edit is not working properly, when I click on the edit button it throws this error in the browser, kindly make sure to fix this so users can modify the tasks

Screenshot_20230623_000423

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @akbarsazish, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @akbarsazish , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Works add, edit & remove✔️
- Well structured Code ✔️
- Used GitHub flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**